indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -237,18 +242,21 @@ public void onFailure(Throwable t)
   }
 
   @LifecycleStop
-  public void stop()
+  public void stop() throws InterruptedException
   {
     synchronized (lifecycleLock) {
       Preconditions.checkState(started, "not started");
 
       try {
+        log.info("Stopping WorkerTaskMonitor.");
+        started = false;
         exec.shutdownNow();
         pathChildrenCache.close();
         taskRunner.stop();
-
-        started = false;
-        log.info("Stopped WorkerTaskMonitor.");
+        doneStopping.await();
```

Comment:
I think there should be a way for someone to be able to break out of a broken latch. I thiiiiink the user should be able to send another -15 to get this to interrupt? 

If so then I think a simple log before this await indicating that you are about to wait for shutdown to complete should suffice.